sstables/sstables.cc

```diff
@@ -1900,14 +1900,36 @@ sstable_writer sstable::get_writer(const schema& s, uint64_t estimated_partition
 
 future<uint64_t> sstable::validate(reader_permit permit, abort_source& abort,
         std::function<void(sstring)> error_handler, sstables::read_monitor& monitor) {
+    auto handle_sstable_exception = [&error_handler](const auto& e, auto& errors) -> std::exception_ptr {
```

Comment:
Please avoid auto parameters. They force the reader to reverse-engineer the types from the body.